include/deal.II/grid/grid_tools.h

```diff
@@ -3883,182 +3887,282 @@ namespace GridTools
   }
 
 
-
-  template <typename DataType, typename MeshType>
-  void
-  exchange_cell_data_to_ghosts(
-    const MeshType &                                     mesh,
-    const std::function<std_cxx17::optional<DataType>(
-      const typename MeshType::active_cell_iterator &)> &pack,
-    const std::function<void(const typename MeshType::active_cell_iterator &,
-                             const DataType &)> &        unpack)
+  namespace
```

Comment:
We generally avoid anonymous namespaces in header files. Can you wrap this into an `internal` namespace`?